src/components/app/list/DevtronAppListContainer.tsx

```diff
@@ -14,223 +14,443 @@
+import { App, DevtronAppListProps } from './types'
+import { DEVTRON_NODE_DEPLOY_VIDEO, URLS } from '../../../config'
 import { getAppList } from '../service'
-import { AppListViewType } from '../config'
 import './list.scss'
+import { AppListPayloadType, AppListSortableKeys } from '../list-new/AppListType'
+import NodeAppThumbnail from '../../../assets/img/node-app-thumbnail.png'
+import { ReactComponent as PlayMedia } from '../../../assets/icons/ic-play-media.svg'
+import DeployCICD from '../../../assets/img/guide-onboard.png'
+import { appListLoadingArray, APP_LIST_HEADERS } from '../list-new/Constants'
+import { ReactComponent as ArrowRight } from '../../../assets/icons/ic-arrow-right.svg'
+import { ReactComponent as Arrow } from '../../../assets/icons/ic-dropdown-filled.svg'
+import { ReactComponent as DevtronAppIcon } from '../../../assets/icons/ic-devtron-app.svg'
+import { ReactComponent as ICHelpOutline } from '../../../assets/icons/ic-help-outline.svg'
+import { ReactComponent as Edit } from '../../../assets/icons/ic-settings.svg'
+import { ExpandedRow } from './expandedRow/ExpandedRow'
 
-class DevtronAppListContainer extends Component<AppListProps, AppListState> {
-    abortController: AbortController
-
-    constructor(props) {
-        super(props)
-        this.state = {
-            code: 0,
-            view: AppListViewType.LOADING,
-            errors: [],
-            apps: [],
-            size: 0,
-            sortRule: {
-                key: SortBy.APP_NAME,
-                order: OrderBy.ASC,
-            },
-            showCommandBar: false,
-            offset: 0,
-            pageSize: 20,
-            expandedRow: null,
-            isAllExpanded: false,
-            isAllExpandable: false,
+const DevtronAppList = ({
+    filterConfig,
+    environmentList,
+    namespaceList,
+    appFiltersResponseLoading,
+    syncListData,
+    updateDataSyncing,
+    setCurrentAppName,
+    clearAllFilters,
+    handleSorting,
+    isArgoInstalled,
+    changePage,
+    changePageSize,
+    setAppCount,
+}: DevtronAppListProps) => {
+    const history = useHistory()
+
+    const [expandedState, setExpandedState] = useState<{
+        expandedRow: Record<number, boolean>
+        isAllExpanded: boolean
+        isAllExpandable: boolean
+    }>({
+        expandedRow: {},
+        isAllExpanded: false,
+        isAllExpandable: false,
+    })
+
+    const { searchKey, offset, pageSize, appStatus, project, environment, namespace, cluster, sortBy, sortOrder } =
+        filterConfig
+
+    const isSearchOrFilterApplied =
+        searchKey || appStatus.length || project.length || environment.length || namespace.length || cluster.length
+
+    const appListPayload: AppListPayloadType = getDevtronAppListPayload(filterConfig, environmentList, namespaceList)
+
+    const abortControllerRef = useRef(new AbortController())
+    const [appListLoading, appListResponse, appListError, appListReload] = useAsync(
```

Comment:
Can loading and error variables to include getIsRequestAborted at variable level itself.